chore(Tooltip): converted to int test

shani-terminus opened the pull request:
ISSUES CLOSED: #1296

shani-terminus committed:
chore(Tooltip): converted to int test

ISSUES CLOSED: #1296

benjamincharity commented on terminus-ui/tooltip/src/tooltip.component.spec.ts:
```diff
@@ -1,46 +1,139 @@
 
-    });
+        expect(tooltipComponent.tooltipPosition).toEqual('above');
+        expect(tooltipEl.attributes.getNamedItem('ng-reflect-position').value).toEqual('above');
```
I don't think we should be testing `ng-*` data attributes directly. We have no control over those values. We should probably be testing for positions or classes. (same comment below)

benjamincharity commented on terminus-ui/tooltip/src/tooltip.component.html:
```diff
@@ -1,8 +1,9 @@
   [matTooltip]="tooltipValue"
   [matTooltipPosition]="tooltipPosition"
   [ngClass]="{'c-tooltip--underline': hasUnderline}"
+  (click)="tooltip.show()"
```
We shouldn't be changing how the tooltip works for consumers. I'd rather add a `show` method to our tooltip component that calls Material's `show`.

benjamincharity commented on terminus-ui/tooltip/testing/src/test-helpers.ts:
```diff
@@ -0,0 +1,17 @@
+ * @param fixture - the fixture with a tooltip
+ */
+// tslint:disable-next-line:no-any
+export function showTooltip(fixture: ComponentFixture<any>) {
```
Need `void` return type here

benjamincharity committed:
chore(Packages): upgrade all packages

angular v8, typescript ~3.4.2

BREAKING CHANGE:
Angular v8 is now required. TypeScript 3.4.x is now required.

ISSUES CLOSED: #1540

benjamincharity commented on terminus-ui/tooltip/testing/src/test-helpers.ts:
```diff
@@ -0,0 +1,20 @@
+import { ComponentFixture } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { MatTooltip } from '@angular/material';
```
I don't think this one is being used anymore

benjamincharity commented on terminus-ui/tooltip/testing/src/test-helpers.ts:
```diff
@@ -0,0 +1,20 @@
+import { MatTooltip } from '@angular/material';
+
+
+// @ViewChild(MatTooltip) tooltip:MatTooltip;
```
Legacy comment

benjamincharity commented on terminus-ui/tooltip/testing/src/test-components.ts:
```diff
@@ -0,0 +1,69 @@
+import {
+  TsTooltipComponent,
+  TsTooltipModule,
+} from '../../src/tooltip.module';
```
This import should be like an external consumer: `@terminus/ui/tooltip`

shani-terminus replied:
Yes, this was supposed to be temporary

benjamincharity commented on terminus-ui/tooltip/src/tooltip.component.ts:
```diff
@@ -79,4 +83,17 @@ export class TsTooltipComponent {
+  /**
+   * Call Material Tooltip's show() method
+   */
+  public showTooltip() {
```
Missing void return type

benjamincharity commented on terminus-ui/tooltip/src/tooltip.component.ts:
```diff
@@ -3,8 +3,12 @@ import {
+  ViewChild,
   ViewEncapsulation,
 } from '@angular/core';
+import { MatTooltip } from '@angular/material';
```
I don't think this one is being used anymore

benjamincharity commented on terminus-ui/button/src/button.component.spec.ts:
```diff
@@ -7,6 +7,7 @@ import {
   tick,
 } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
+import { getDomAttribute } from '@terminus/ngx-tools';
```
Oh, also, this is a rebase issue I think.. you can go ahead and remove this line.

benjamincharity committed:
Merge branch 'release' into 1296-tooltip-convert-tests
